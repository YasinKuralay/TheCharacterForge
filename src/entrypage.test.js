import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import EntryPage from "./entrypage.js";

it("renders the EntryPage component without crashing", () => {
    const div = document.createElement("div");
    // The BrowserRouter is for eliminating the unnecessary error "Don't use Link outside Router" from Jest, which is caused by using an old version of the Router, and will be fixed
    ReactDOM.render(
        <BrowserRouter>
            <EntryPage />
        </BrowserRouter>,
        div
    );
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the previous snapshot of the EntryPage Component", () => {
    const EntryPageComponentSnapshot = renderer
        .create(
            <BrowserRouter>
                <EntryPage />
            </BrowserRouter>
        )
        .toJSON();
    expect(EntryPageComponentSnapshot).toMatchSnapshot();
});
