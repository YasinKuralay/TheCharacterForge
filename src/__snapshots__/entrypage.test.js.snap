// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot of the EntryPage Component 1`] = `
<div>
  <a
    href="/browseCharacters"
    onClick={[Function]}
  >
    <div
      id="browseCharacters"
    >
      <h2>
        Browse Your Characters
      </h2>
      <i
        className="fas fa-search fa-3x"
      />
    </div>
  </a>
  <a
    href="/createcharacter"
    onClick={[Function]}
  >
    <div
      id="createCharacters"
    >
      <h2>
        Create New Character
      </h2>
      <i
        className="fas fa-hat-wizard fa-3x"
      />
    </div>
  </a>
</div>
`;
